### PR TITLE
Improve resource hints

### PR DESCRIPTION
**Pull request by eagerterrier**

### CODE REVIEW

File: src/server/index.jsx [hunk 1]
```diff
@@ -134,6 +133,20 @@ const injectDefaultCacheHeader = (req, res, next) => {
...
+const injectResourceHintsHeader = (req, res, next) => {
+  if (req.originalUrl.indexOf('/pidgin') === 0) {
+    const assetOrigins = getAssetOrigins('pidgin');
+    res.set(
+      'Link',
+      assetOrigins
+        .reduce((toReturn, domainName) => {
+          return `${toReturn}<${domainName}>; rel="dns-prefetch", <${domainName}>; rel="preconnect",`;
+        }, '')
+        .slice(0, -1),
```
Comment (amoore108) [inline src/server/index.jsx]: Would something like this work? Little bit easier to read:
```suggestion
        assetOrigins.map(
          domainName =>`<${domainName}>; rel="dns-prefetch", <${domainName}>; rel="preconnect"`,
        ).join(',')
```

**Review by amoore108 — APPROVED** [review]

File: src/server/index.jsx [hunk 2]
```diff
@@ -134,6 +133,18 @@ const injectDefaultCacheHeader = (req, res, next) => {
   next();
 };
 
+const injectResourceHintsHeader = (req, res, next) => {
+  if (req.originalUrl.indexOf('/pidgin') === 0) {
```
Comment (jroebu14) [inline src/server/index.jsx]: woulkd this work?

```suggestion
  const isPidginService = req.originalUrl.startsWith('/pidgin')
  
  if (isPidginService) {
```

File: src/server/index.jsx [hunk 3]
```diff
@@ -134,6 +133,18 @@ const injectDefaultCacheHeader = (req, res, next) => {
...
+    const assetOrigins = getAssetOrigins('pidgin');
+    res.set(
+      'Link',
+        assetOrigins.map(
+          domainName =>`<${domainName}>; rel="dns-prefetch", <${domainName}>; rel="preconnect"`,
```
Comment (jroebu14) [inline src/server/index.jsx]: would we need tests for this? I struggle to understand the string it builds. Is it `<bbc.com>; rel="dns-prefetch", <bbc.com>; <bbc.co.uk>; rel="dns-prefetch", <bbc.co.uk>;`? is that right? what is it? should the callback function maybe be abstracted out and given a name?

Comment (eagerterrier) [reply src/server/index.jsx]: look like this - ```<http://static.files.bbci.co.uk>; rel="dns-prefetch", <http://static.files.bbci.co.uk/>; rel="preconnect", <http://news.files.bbci.co.uk>; rel="dns-prefetch", <http://news.files.bbci.co.uk/>; rel="preconnect"```

could be abstracted, but is pretty small and is just a small experiment for pidgin a/t/m, so I figure we could run with it for now

**Pull request merged**